### ISSUE TITLE
Add more data sanity checks

### ISSUE THREAD

**Issue by piskvorky** [RaRe-Technologies/gensim#74]
There's been a steady trickle of reports that LSI/LDA misbehave, produce degenerate models, crash Python etc. 

Typically this is a user data problem (bad input data, feature id mismatch, ...), but since gensim targets the wide general public, this is gensim's "fault" anyway.

Create utility functions that perform basic sanity checks on user's input data:
1. check that the all feature ids in a corpus are compatible with the user-provided dictionary (should avoid issues like http://projects.scipy.org/scipy/ticket/1582 )
2. check that the data range is valid -- look for NaNs, Infs, explicit zeros => these are all illegal in gensim input.
3. check that the data is not degenerate => all vectors identical/empty/?/model looks weird
4. ~check corpus type and warn the user if it's plain list (promote the memory-friendly generator interface, shown in tutorials)~ NOT NEEDED

**Comment by prakhar2b** [comment]
@tmylk I would like to work on this. @piskvorky the link above is broken, could you brief me what it was about ?

**Comment by piskvorky** [comment]
@prakhar2b I think it was a scipy crash (segfault) when using sparse arrays and indexing an element out-of-bounds.

I wouldn't say this issues is "easy" -- it will need some careful thinking and planning. We definitely don't want to slow down processing too much, by (for example) requiring an extra data pass just to check for bad values.

**Comment by rasto2211** [comment]
I would like to work on this issue. Could you please give me some pointers to the code where to start?

**Comment by tmylk** [comment]
@rasto2211  Adding a warning to `LdaModel.init` when the input is a list is a good way to start ( item 4 above)

**Comment by rasto2211** [comment]
@menshikh-iv  Do you also want to close this issue since you closed my PR without merging?

**Comment by menshikh-iv** [comment]
@rasto2211 No, because the remaining points are important (see [radim comment](https://github.com/RaRe-Technologies/gensim/pull/1392#issuecomment-308373375))

**Comment by menshikh-iv** [comment]
Be attentive with #1732, I already see exactly same problem twice
